modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js

```diff
@@ -50,23 +51,55 @@ function loadAvailableJavaPolicies() {
     });
 }
 
-function updateJavaPolicyPartial(){
+/**
+ * draw java policies list
+ */
+function updateJavaPolicyPartial() {
+    var enabledStatus;  //if policy is mandatory make it wont allow user to un-tick the particular java policy
+    var checkedStatus; //if policy is mandatory always tick the java policy
 
-    //draw java policies list
     $.each(javaPolicyArray, function (index, obj) {
         if (obj != null) {
-            console.info(obj)
+
+            if (obj.isMandatory) {
+                //always check and and disable the mandatory fields
+                enabledStatus = 'disabled';
+                checkedStatus = 'checked';
+            } else {
+                enabledStatus = 'enabled';
+
+                //tick the saved mappings
+                if (obj.applicationId == null) {
+                    checkedStatus = "";
+                }
+                else {
+                    checkedStatus = "checked";
+                }
+            }
+
             $('#javaPolicy_tbody').append(
-                '<tr>' +
-                '<td>' + obj.displayName + '</td>' +
-                '<td>' + obj.description + '</td>' +
-                '<td><input data-javaPolicy-id=' + obj.id + ' type=checkbox disabled checked></td>' +
-                '</tr>');
+                "<tr>" +
+                "<td>" + obj.displayName + "</td>" +
+                "<td>" + obj.description + "</td>" +
+                "<td><input  class='javaPolicy-opt-val' data-javaPolicy-id='" + obj.javaPolicyId +
+                "'  type='checkbox' " + enabledStatus +
+                " " + checkedStatus + "></td>" +
+                "</tr>");
 
-            //policyGroupIndexArray.push(obj.policyGroupId);
         }
     });
 
     //draw java policy items
     $("#javaPolicy_tbody").trigger("draw");
+
+    //update the url pattern wise policy group drop downs
+    $('.javaPolicy-opt-val').each(function () {
```

Comment:
This code block does nothing, other than having a TODO comment. Remove this code block and add a descriptive TODO if it is really needed.